### PR TITLE
feat(marketplaces): approve spending function

### PR DESCRIPTION
**Pull request by vlnevyhosteny**
Add missing approve spending function to marketplace listing.

### CODE REVIEW

**Comment by pfranc** [comment]
Task linked: [PLT-373 [SDK] Complete function - Marketplaces](https://app.clickup.com/t/24443045/PLT-373)

**Comment by npwork** [comment]
Please rebase

**Comment by vlnevyhosteny** [comment]
@npwork done

File: packages/shared/testing/evm-based/src/lib/evm-based.marketplace.test-factory.ts [hunk 1]
```diff
@@ -273,5 +273,65 @@ export const marketplaceTestFactory = {
...
+            fromPrivateKey: '0x05e150c73f1920ec14caa1e0b6aa09940899678051a78542840c2668ce5080c2',
+            nonce: 1,
+            fee: {
+              gasLimit: '40000',
+              gasPrice: '20',
+            },
+            amount: '10000',
+          }),
+        ).rejects.toThrow(
+          `Provided address 0x687422eEA2cB73B5d3e242bA5456b782919AFc86 is invalid, the capitalization checksum test failed, or it's an indirect IBAN address which can't be converted.`,
```
Comment (npwork) [inline packages/shared/testing/evm-based/src/lib/evm-based.marketplace.test-factory.ts:332]: Can we reuse SDK errors for that?

Comment (vlnevyhosteny) [reply packages/shared/testing/evm-based/src/lib/evm-based.marketplace.test-factory.ts:332]: I dont think so because only string is thrown. 😕

Comment (jdckl) [reply packages/shared/testing/evm-based/src/lib/evm-based.marketplace.test-factory.ts:332]: Let's add it to the sdk errors constants?

**Review by jdckl — APPROVED** [review]

**Pull request merged**